src/ru/waymanga/src/eu/kanade/tachiyomi/extension/ru/waymanga/WayManga.kt

```diff
@@ -0,0 +1,145 @@
+package eu.kanade.tachiyomi.extension.ru.waymanga
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.net.URLEncoder
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class WayManga : ParsedHttpSource() {
+
+    override val name = "WayManga"
+
+    override val baseUrl = "https://waymanga.ru"
+
+    override val lang = "ru"
+
+    private val cookiesHeader by lazy {
+        val cookies = mutableMapOf<String, String>()
+        cookies["ageRestrict"] = "17"
+        buildCookies(cookies)
+    }
+
+    private fun buildCookies(cookies: Map<String, String>) =
+        cookies.entries.joinToString(separator = "; ", postfix = ";") {
+            "${URLEncoder.encode(it.key, "UTF-8")}=${URLEncoder.encode(it.value, "UTF-8")}"
+        }
+
+    override val client = network.client.newBuilder()
+        .addNetworkInterceptor { chain ->
+            val newReq = chain
+                .request()
+                .newBuilder()
+                .addHeader("Cookie", cookiesHeader)
+                .build()
+
+            chain.proceed(newReq)
+        }.build()!!
```

Comment:
This seems unnecessarily complicated for no reason.

```suggestion
    override fun headersBuilder() =
        super.headersBuilder().add("Cookie", "ageRestrict=17;")
```